lib/ui/screens/album_screen.dart

```diff
@@ -49,39 +65,26 @@ class AlbumScreen extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return BlocProvider(
-      create: (context) => AlbumCubit(
-        RepositoryProvider.of<ApiRepository>(context),
-      )..load(slug),
-      child: BlocBuilder<AlbumCubit, AlbumScreenState>(
+    return BlocProvider.value(
+      value: _cubit,
+      child: BlocBuilder<AlbumCubit, AlbumState>(
         builder: (context, state) {
           late final Widget body;
-          if (state.isLoading) {
-            body = const Center(child: CircularProgressIndicator());
-          } else if (state.hasException) {
+          if (state is ResultState) {
```

Comment:
Maybe use a typeswitch for this, it doesnt look idea in dart but its nicer than if else

Reply:
That would be semantically different with subtyping (runtimetype does not match parent types). I think it'd be a good idea to find a consistent way to do this but I think this is fine for now. Perhaps later I can either use runtime type everywhere or add a helper method that swwitches between a number of closures:
```
state.switch(
  onLoading: (LoadingState state) {...},
  onError: (ErrorState state) {...},
  onResult: (ResultState state) {...},
  [else: (state) {...}],
)
```

Reply:
I also like the idea of handling a lot of the common screen in a dedicated widget but we can discuss this next time not very important for this MR